drivers/i2c/Kconfig

```diff
@@ -56,6 +56,7 @@ config I2C_CALLBACK
 
 config HAS_I2C_RTIO
 	bool
+	default y
```

Comment:
This breaks the reason for having this imo, or at the very least we should have Kconfig markers for drivers that natively support the feature or not. Having this always y defeats the intended purpose of being able to do these checks at compile time.

Perhaps better would be a per DT node option though? Devices might have multiple i2c controllers, some may or may not support the feature natively.